src/track/beatstats.cpp

```diff
@@ -0,0 +1,84 @@
+#include <QList>
+#include <QMap>
+
+#include <algorithm>
+#include <cmath>
+
+#include "track/beatstats.h"
+#include "util/fpclassify.h"
+
+
+void MovingAvarage::update(double newValue, double oldValue) {
+    if (!util_isnan(oldValue)) {
+        m_sum -= oldValue;
+    }
+    m_sum += newValue;
+}
+
+void MovingAvarage::reset() {
+    m_sum = 0;
+    clear();
+}
+
+double MovingAvarage::compute() {
+    return m_sum / static_cast<double>(windowSize());
+}
+
+void MovingMode::update(double newValue, double oldValue) {
+    m_tempoFrequency[newValue] += 1;
```

Comment:
It only works because all tempo values pushed into this are rounded to 2 decimals only.

Reply:
Please explain as a code comments. 
Can we rely on this? I am in doubt. 

We have already quantize beat length in the original complex SD.
I think it would be more reliable to switch to integer numbers counting these samples.

Reply:
I think it`s reliable. It's the same approach used all over beatutils all over for almost a decade.

Reply:
I think this is only true because the values are quantized already bx our 0,011 Ms sample rate.

After any ironing stage this will be broken.
I think we need to add something that verifies this.

Can we express that in a sample rate or precision parameter?

Reply:
Won't the sample rate have the same problem since our samples are in double too and we allow fractional sample positions?
A precision parameter will not work here because the comparison is made by qt container itself.
Unless we add a class wrapper and overload ==
Should do it like that or do you have better ideas?